### PR TITLE
Initial commit of Gen App Builder (Chat App)

### PR DESCRIPTION
**Pull request by koverholt**
This PR adds:

- Initial site for Gen App Builder (Chat App) - site is built w/ Svelte and Flowbite-Svelte
- Data store indexes Google Store
- Includes benefits and how to use the demo chat app
- Includes details of how chat app was built (will add more to this w/ diagrams)

Preview of site in current state: https://gen-app-builder-chat.web.app/

I wanted to first get the backend agent working and a skeleton site. Then, in a followup PR, I will clean up and prettify the app frontend, and add more documentation to the README.

Remaining TODOs in this PR:
- [x] Add license headers to relevant files
- [x] Fix linting / formatting errors

### CODE REVIEW

File: gen-app-builder/chat-app/site/.firebaserc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "projects": {
+    "default": "koverholt-devrel-355716"
```
Comment (holtskinner) [inline gen-app-builder/chat-app/site/.firebaserc:3]: Remove References to your own project, maybe replace with `YOUR_PROJECT_ID`

Also add steps to readme to replace those values in specific filenames

File: gen-app-builder/chat-app/site/README.md
Comment (holtskinner) [inline gen-app-builder/chat-app/site/README.md:1]: Maybe get rid of this file since it seems to be a template?

**Review by holtskinner — APPROVED** [review]
LGTM overall.

Can you update the repo README to add a link to this directory?

General question, does the `site/` directory need to exist? Could all of the files be moved up into the `chat-app` directory?

File: gen-app-builder/chat-app/site/src/routes/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    <p class="text-lg">
+      <b>Step 1:</b> Create a chat app and associated data store in Gen App Builder <br /><br />
+      <b>Step 2:</b> Select the website data type and input your website URLs <br /><br />
+      <b>Step 3:</b> Embed a chat widget into your website and delight your customers!
+    </p>
+  </TabItem>
+</Tabs>
+
+Powered by <a href="https://cloud.google.com/generative-ai-app-builder">Gen App Builder</a> and
+<a href="https://cloud.google.com/dialogflow">Dialogflow CX</a>
```
Comment (holtskinner) [inline gen-app-builder/chat-app/site/src/routes/+page.svelte:103]: Change the styling for these to indicate that they are links.

File: gen-app-builder/chat-app/site/src/routes/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        </blockquote>
+      </figure>
+    </Card>
+  </TabItem>
+  <TabItem class="w-full">
+    <span slot="title">How we built it</span>
+    <p class="text-lg">
+      <b>Step 1:</b> Create a chat app and associated data store in Gen App Builder <br /><br />
+      <b>Step 2:</b> Select the website data type and input your website URLs <br /><br />
+      <b>Step 3:</b> Embed a chat widget into your website and delight your customers!
```
Comment (holtskinner) [inline gen-app-builder/chat-app/site/src/routes/+page.svelte:97]: > delight your customers!

🙃🙃🙃

File: gen-app-builder/chat-app/site/README.md
Comment (koverholt) [reply gen-app-builder/chat-app/site/README.md:1]: Done!

File: gen-app-builder/chat-app/site/src/routes/+page.svelte (hunk 2 above)
Comment (koverholt) [reply gen-app-builder/chat-app/site/src/routes/+page.svelte:103]: Will address this in a future beautification PR and make it way better. 😁

File: gen-app-builder/chat-app/site/src/routes/+page.svelte (hunk 3 above)
Comment (koverholt) [reply gen-app-builder/chat-app/site/src/routes/+page.svelte:97]: 😅😅😅

File: gen-app-builder/chat-app/site/.firebaserc (hunk 1 above)
Comment (koverholt) [reply gen-app-builder/chat-app/site/.firebaserc]: Thanks for catching this. I generalized the values in 356e5cf.

Normally I don't commit .rc files and such, but committing it here captures a potentially confusing step when deploying to firebase. Nonetheless, I will give this a couple of more passes when writing up the docs in the README and make this as easy/clean as possible.

**Comment by koverholt** [comment]
> General question, does the site/ directory need to exist? Could all of the files be moved up into the chat-app directory?

Thanks for noting this, it's mostly muscle memory as I typically put the backend code (or an exported Dialogflow CX agent) and other services in other folders, sort of like a breakdown by microservice. But since the site/app is the main deliverable here, I moved all of the content of `site/` up one level in b76d442.

**Comment by koverholt** [comment]
> Can you update the repo README to add a link to this directory?

Thanks for the reminder, added!

**Comment by koverholt** [comment]
All done with this PR. Thanks for the review, @holtskinner. Lots of more beautification to the app and README docs/links coming in a followup PR soon!

**Pull request merged**